standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/ReplChangeManager.java

```diff
@@ -356,6 +356,9 @@ public static FileInfo getFileInfo(Path src, String checksumString, String srcCM
   // Currently using fileuri#checksum#cmrooturi#subdirs as the format
   public static String encodeFileUri(String fileUriStr, String fileChecksum, String encodedSubDir)
           throws IOException {
+    if (instance == null) {
```

Comment:
It can be assert as we expect caller to always instantiate ReplChangeManager before invoking this method.
Also, if you think, exception is better, then shall use IllegalStateException to be relevant.

Reply:
assert may not trigger in production and we won't catch the problem in REPL DUMP itself. The problem will be seen only in REPL LOAD, and that too with no clue as to what went wrong. So, it's better to have an exception. IllegalStateException is better. Done.